csharp/src/Ice/Communicator.cs

```diff
@@ -430,8 +430,10 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
 
                 DefaultTransport = GetProperty("Ice.Default.Transport") ?? "tcp";
 
-                DefaultInvocationTimeout = GetPropertyAsInt("Ice.Default.InvocationTimeout") ?? -1;
-                if (DefaultInvocationTimeout < 1 && DefaultInvocationTimeout != -1)
+                DefaultInvocationTimeout =
+                    GetPropertyAsTimeSpan("Ice.Default.InvocationTimeout") ?? Timeout.InfiniteTimeSpan;;
+
+                if (DefaultInvocationTimeout != Timeout.InfiniteTimeSpan && DefaultInvocationTimeout <= TimeSpan.Zero)
```

Comment:
I think we can remove this check `GetPropertyAsTimeSpan` already rejects negative values, it accepts positive integers or `infinite`